surfsense_backend/app/tasks/background_tasks.py

```
@@ -327,20 +349,16 @@ def get_youtube_video_id(url: str):
+            "format": "json",
+            "url": f"https://www.youtube.com/watch?v={video_id}",
+        }
         oembed_url = "https://www.youtube.com/oembed"
-        query_string = urlencode(params)
-        full_url = oembed_url + "?" + query_string
 
-        with urlopen(full_url) as response:
-            response_text = response.read()
-            video_data = json.loads(response_text.decode())
+        async with aiohttp.ClientSession() as session:
+            async with session.get(oembed_url, params=params) as response:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)